### PR TITLE
raise error for sidecar or popout when not supported

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request raises an error when trying to show a widget in popout or sidecar when they are not supported (i.e. in VS Code).  Related to #3251.

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/core/template_mixin.py [hunk 1]
```diff
@@ -126,6 +127,8 @@ def show_widget(widget, loc, title):  # pragma: no cover
         display(widget)
 
     elif loc.startswith('sidecar'):
+        if not get_kernel_id():
```
Comment (kecnry) [inline jdaviz/core/template_mixin.py:130]: @mariobuikhuizen - is this sufficient, or would accessing `get_base_url` be more appropriate (in which case can that be exposed back to python in ipypopout)?

Comment (mariobuikhuizen) [reply jdaviz/core/template_mixin.py:130]: I don't think there is a 1 on 1 mapping of where ipypopout and sidecar work. I think sidecar has different requirements than ipypopout.

Comment (kecnry) [reply jdaviz/core/template_mixin.py:130]: do you have any suggestions on which conditions to use for each for now then?

**Comment by rosteen** [comment]
The error messages get triggered for me as expected. I noticed one or two other interesting things happening when running in VSCode, do we want to (or did we already?) add a disclaimer in the docs that running in VSCode is not fully supported?

**Comment by gibsongreen** [comment]
I was also able to reproduce the error for the default python environment and in the case of if I set it to my normal work environment. 

I did find a case that if you have a jupyter server running independent of VSCode, you can select the option for an existing Jupyter server, which then will result in an existing kernel id and the error does not get raised.

**Comment by rosteen** [comment]
> @rosteen - can you please create tickets for anything you found? We can then discuss whether we should add a warning or not (I don't think we advertise anywhere that we _do_ support VSCode, but if its sufficiently broken, we should either fix it or maybe make a note in the docs).

See #3313, I'll see if I can find anything else.

File: jdaviz/core/template_mixin.py (hunk 1 above)
Comment (mariobuikhuizen) [reply jdaviz/core/template_mixin.py:130]: I don't know how to detect if sidecar is available.

Comment (kecnry) [reply jdaviz/core/template_mixin.py:130]: ok, then I think this is better than nothing for now and we can always revise the condition going forward 🤷‍♂️

**Review by gibsongreen — APPROVED** [review]
With the conversation in mind, this looks good to me! 

I added a video to #3313 with additional related behavior. In the video it also shows  unrelated issues in which icons are unable to display throughout the app and axes become difficult to read in dark-mode. This didn't occur when testing Imviz and when I tested before in Cubeviz I saw both expected and bug behavior. I'm in favor in a separate effort of adding a docs disclaimer along with a note for the user to open issues.

**Pull request merged**